### PR TITLE
Add documentation regarding transactional exceptions

### LINKED ISSUE #2290: Document exceptions for retryable transactions
As of https://github.com/doctrine/dbal/pull/718 `2.6` will receive new exceptions for retryable transactions. This feature must get a little chapter in the documentation about transactions before release.

### PR DESCRIPTION
**Pull request by dsantang**
Fixes #2290

### CODE REVIEW

File: docs/en/reference/transactions.rst [hunk 1]
```diff
@@ -235,3 +235,27 @@ To initialize a ``Doctrine\DBAL\Connection`` with auto-commit disabled,
...
 and pass it to a ``Doctrine\DBAL\Connection`` when instantiating.
+
+
+Error handling
+--------------
+
+If you want to handle errors related to deadlocks or lock wait timeouts you can use 
+
+It is best practice to implement retry logic for transactions that are aborted because of deadlocks or timeouts.
+Doctrine provides built-in exceptions for these kind of errors.
```
Comment (Ocramius) [inline docs/en/reference/transactions.rst]: `these kindS`

File: docs/en/reference/transactions.rst [hunk 2]
```diff
@@ -235,3 +235,27 @@ To initialize a ``Doctrine\DBAL\Connection`` with auto-commit disabled,
 you can also use the ``Doctrine\DBAL\Configuration`` container to modify the
 default auto-commit mode via ``Doctrine\DBAL\Configuration::setAutoCommit(false)``
 and pass it to a ``Doctrine\DBAL\Connection`` when instantiating.
+
+
+Error handling
+--------------
+
+If you want to handle errors related to deadlocks or lock wait timeouts you can use
```
Comment (Ocramius) [inline docs/en/reference/transactions.rst]: Sentence is cut off

File: docs/en/reference/transactions.rst [hunk 3]
```diff
@@ -235,3 +235,27 @@ To initialize a ``Doctrine\DBAL\Connection`` with auto-commit disabled,
...
+
+Error handling
+--------------
+
+If you want to handle errors related to deadlocks or lock wait timeouts you can use 
+
+It is best practice to implement retry logic for transactions that are aborted because of deadlocks or timeouts.
+Doctrine provides built-in exceptions for these kind of errors.
+
+Transaction exceptions have a marker interface: ``Doctrine/DBAL/Exception/RetryableException`` and based on the .
```
Comment (Ocramius) [inline docs/en/reference/transactions.rst]: `on the .`?

File: docs/en/reference/transactions.rst [hunk 4]
```diff
@@ -235,3 +235,27 @@ To initialize a ``Doctrine\DBAL\Connection`` with auto-commit disabled,
...
+Error handling
+--------------
+
+If you want to handle errors related to deadlocks or lock wait timeouts you can use 
+
+It is best practice to implement retry logic for transactions that are aborted because of deadlocks or timeouts.
+Doctrine provides built-in exceptions for these kind of errors.
+
+Transaction exceptions have a marker interface: ``Doctrine/DBAL/Exception/RetryableException`` and based on the . 
+A practical example is as follows:
```
Comment (Ocramius) [inline docs/en/reference/transactions.rst]: Specify what kind of example this is - what is it for?

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by dsantang** [comment]
my bad, amended with the real changes.

**Review by Ocramius — APPROVED** [review]

**Pull request merged**